File: src/main/java/it/unibo/crabinv/Controller/Entity/EntityController.java
package it.unibo.crabinv.Controller.Entity;

import it.unibo.crabinv.Model.entity.Delta;

/**
 * Provides methods to control an entity standardizing the way they should be controlled
 */
public interface EntityController {
    /**
     * Updates the status of something that receives inputs, such as the player
     * @param firePressed
     * @param delta
     */
    void update(boolean firePressed, Delta delta);

    /**
     * Updates the status of something that doesn't receive inputs,
     * and therefore has constant movement
     */
    void update(Delta delta);
}

File: src/main/java/it/unibo/crabinv/controller/entities/entity/EntityController.kt
package it.unibo.crabinv.controller.entities.entity

/**
 * Provides methods to control an entity standardizing the way they should be controlled.
 */
interface EntityController {
    /**
     * @return if the entity is alive
     */
    val isAlive: Boolean

    /**
     * Make the entity suffer the inputted amount of damage.
     * @param damage the damage the entity should suffer
     */
    fun takeDamage(damage: Int)

    /**
     * @return the current health of the entity
     */
    val health: Int

    /**
     * @return the max health of the entity
     */
    val maxHealth: Int

    /**
     * @return the x coordinate of the entity
     */
    val x: Double

    /**
     * @return the y coordinate of the entity
     */
    val y: Double
}
